srtcore/epoll.cpp

```diff
@@ -478,6 +495,47 @@ int CEPoll::wait(const int eid, set<UDTSOCKET>* readfds, set<UDTSOCKET>* writefd
                ++ total;
             }
          }
+         #elif defined(WIN32)
+
+         DWORD ready;
+         DWORD timeout_ms = 10;
+         const int nhandles = p->second.m_mLocals.size();
+         HANDLE handles[MAXIMUM_WAIT_OBJECTS];
+         int i = 0;
+
+         for (std::map<SYSSOCKET, CEPollFD>::iterator it = p->second.m_mLocals.begin();
+              it != p->second.m_mLocals.end();
+              ++it)
+         {
+            handles[i++] = (HANDLE) it->first;
+         }
+
+         ready = WaitForMultipleObjectsEx(nhandles, handles, FALSE, timeout_ms, TRUE);
+
+         if (ready == WAIT_FAILED)
+         {
+            throw CUDTException(MJ_NOTSUP, MN_INVAL, 0);
+         }
+
+         if (ready >= WAIT_OBJECT_0 && ready < WAIT_OBJECT_0 + nhandles)
+         {
+            SYSSOCKET s = *((int *)&handles[ready - WAIT_OBJECT_0]);
+            CEPollFD pollFD = p->second.m_mLocals[s];
+            pollFD.revents = pollFD.events;
+
+            if ((NULL != lrfds) && (pollFD.events & UDT_EPOLL_IN))
```

Comment:
Why go through the trouble of figuring out what direction user requested us to notify about if we can't figure that out using WIN32 events and WaitForMultipleObjectsEx()? I feel like you are mixed up between events and I/O objects.